token/src/instruction.rs

```diff
@@ -51,6 +66,7 @@ pub enum TokenInstruction {
     ///   1. `[writable]` Source/Delegate account.
     ///   2. `[writable]` Destination account.
     ///   3. Optional: `[writable]` Source account if key 1 is a delegate account.
+    ///   4-14. Optional: `[Signer]` M multisignature Signer accounts
```

Comment:
Doesn't the `0. `[signer]` Owner of the source account.` account disappear?
The single owner concept is implicit 1x1 multisigner account, so it could just be included as the first multisignature Signer account.

That `3. Optional: `[writable]` Source account if key 1 is a delegate account.` account is now clunky.  I guess it goes away entirely when we drop from supporting N delegate accounts to just 1.

Reply:
I was planning to keep the single owner support as is which has a shorter code path and less required keys then having to implore the multi-sig route.

Yeah, I thought clunky too, and left it to get suggestions.  It would go away if we do single delegate but it's indicative of a clunkiness we will have with other programs in the future especially one's that support a group of optional sigs as multisig support does.

Reply:
Keeping single owner support makes sense to me.   I'd think that from the interface perspective, using a single owner keypair would look identical to using a `Multisig` account configured at 1x1

Reply:
Hmm, it looks like with this approach, the multisig account itself always has to be a signer (in order for it to appear as the first account in the metas list and be marked as the owner). That doesn't seem right to me.

Reply:
Should the accounts maybe be something more like this?
```
*  (Multiple) `[signer]` Owner of the source account or M multisignature Signer accounts
1. `[writable]` Source/Delegate account.
2. `[writable]` Destination account.
3. (Optional) `[writable]` Source account if key 1 is a delegate account.
4. (Optional) `[]` Multisignature data account
```

Reply:
I was thinking:
```
    ///   0. `[writable]` Source/Delegate account.
    ///   1. `[writable]` Destination account.
    ///   2. Optional: `[writable]` Source account if key 1 is a delegate account.
    ///   2-12, or 3-13. `[Signer]` M multisignature Signer accounts (1 if a single owner address)
```

Reply:
Oh wait that doesn't work

Reply:
Yeah, so the complication of storing the multisig data in a separate account is that its address has to be passed in every time

Reply:
Assuming we drop the N delegate support, and each token only supports one delegate:

Single owner/delegate:
```
    ///   0. `[writable]` Source account.
    ///   1. `[writable]` Destination account.
    ///   2. `[Signer]` Owner/delegate address
```

Multisig owner/delegate:
```
    ///   0. `[writable]` Source account.
    ///   1. `[writable]` Destination account.
    ///   2. `[]` Multisig account
    ///   3-M. `[Signer]` M multisignature Signer accounts
```

Reply:
Good point @CriesofCarrots, if its a multisig owner then the multisig does not need to be a signer.  I think this is only an issue in the instruction constructors functions as currently implemented.  Do you see the issue manifest beyond those constructors and the current instruction docs?

Reply:
The single delegate simplifies things a lot.  I like your suggestion @mvines, removing the "optional" from the docs makes things more clear.  For Multisig owner/delegate we should to clarify that the multisig account can either be the owner or the delegate multisig account since the behavior is different between the two
`2. '[]' Owner/delegate Multisig account`